aten/src/ATen/cudnn/Handle.cpp

```diff
@@ -34,8 +35,67 @@ struct Handle {
 };
 
 std::mutex mutex;
-std::unordered_map<int, Handle> handles;
 
+// Handles are lazily created as different threads request them,
+// but are never destroyed until the end of the program.
+// This is the desired behavior: We want unused handles to stick around,
+// so they can be reused by other threads later.
```

Comment:
I think this comment should explain why handle reuse is important (namely, that if we don't reuse handles, then "bad performance with data parallel which creates new threads for each forward pass")